src/Form/StanfordMediaDialogForm.php

```diff
@@ -0,0 +1,134 @@
+    parent::__construct($entity_repository, $entity_display_repository);
+    $this->dialogPluginManager = $dialog_plugin_manager;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, EditorInterface $editor = NULL) {
+    $form = parent::buildForm($form, $form_state, $editor);
+    $media = $this->getFormMediaEntity($form_state);
+    foreach ($this->getDialogAlterPlugins($media) as $plugin) {
+      $plugin->alterDialogForm($form, $form_state, $editor);
+    }
+    return $form;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+    $media = $this->getFormMediaEntity($form_state);
+    foreach ($this->getDialogAlterPlugins($media) as $plugin) {
+      $plugin->validateDialogForm($form, $form_state);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $response = parent::submitForm($form, $form_state);
+    $commands = $response->getCommands();
+
+    if ($commands[0]['command'] == 'editorDialogSave') {
+      $values = $commands[0]['values'];
+
+      $media = $this->getFormMediaEntity($form_state);
+
+      foreach ($this->getDialogAlterPlugins($media) as $plugin) {
+        $plugin->alterDialogValues($values, $form, $form_state);
+      }
+
+      $response = new AjaxResponse();
+      $response->addCommand(new EditorDialogSave($values));
+      $response->addCommand(new CloseModalDialogCommand());
+    }
+    return $response;
+  }
+
+  /**
+   * Get the entity UUID and load the media entity associated to it.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Current form state.
+   *
+   * @return \Drupal\media\MediaInterface|null
+   *   Media entity in the form.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function getFormMediaEntity(FormStateInterface $form_state) {
+    $media_uuid = $form_state->get(['media_embed_element', 'data-entity-uuid']);
+    return $this->entityRepository->loadEntityByUuid('media', $media_uuid);
+  }
+
+  /**
+   * Get an array of all plugins that are supposed to alter the dialog.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   Media entity.
+   *
+   * @return \Drupal\stanford_media\Plugin\MediaEmbedDialogInterface[]
+   *   Array of plugins.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  protected function getDialogAlterPlugins(MediaInterface $media) {
+    $plugins = [];
+    foreach (array_keys($this->dialogPluginManager->getDefinitions()) as $plugin_id) {
+      /** @var \Drupal\stanford_media\Plugin\MediaEmbedDialogInterface $plugin */
+      $plugin = $this->dialogPluginManager->createInstance($plugin_id, ['entity' => $media]);
+
+      if ($plugin->isApplicable()) {
```

Comment:
This is always true?
https://github.com/SU-SWS/stanford_media/pull/22/files/c5b71881babea53c03775bf8b36f11ac0c780137#diff-05e4b10df2a11ccbb5fb6a3d03df4060R20

Reply:
For that particular plugin. But not all plugins https://github.com/SU-SWS/stanford_media/blob/drupal-8.8-upgrade/src/Plugin/MediaEmbedDialog/File.php#L21